go_modules/lib/dependabot/go_modules/update_checker/latest_version_finder.rb

```diff
@@ -158,6 +159,41 @@ def fetch_latest_version(language_version: nil)
           candidate_versions.max_by(&:version)&.version
         end
 
+        sig do
+          params(releases: T::Array[Dependabot::Package::PackageRelease])
+            .returns(T::Array[Dependabot::Package::PackageRelease])
+        end
+        def filter_incompatible_versions(releases)
```

Comment:
Should we check whether `@goprivate` is present and skip this filter if it is?

We can mention in a comment what `GOPRIVATE` is, and that the `@goprivate` variable will be set if there is a `git_registry` defined in the `dependabot.yaml` and link to _mctofu's post